imohash/imohash.py

```diff
@@ -12,18 +12,23 @@
 SAMPLE_THRESHOLD = 128 * 1024
 SAMPLE_SIZE = 16 * 1024
 
-def hashfile(filename, sample_threshhold=SAMPLE_THRESHOLD, sample_size=SAMPLE_SIZE, hexdigest=False):
-    size = os.path.getsize(filename)
-
-    with open(filename, 'rb') as f:
-        if size < sample_threshhold or sample_size < 1:
-            data = f.read()
-        else:
-            data = f.read(sample_size)
-            f.seek(size//2)
-            data += f.read(sample_size)
-            f.seek(-sample_size, os.SEEK_END)
-            data += f.read(sample_size)
+#Hashes an opened file object. Compatible with paramimo SFTPFile and regular files.
+def hashfileobject(f, sample_threshhold=SAMPLE_THRESHOLD, sample_size=SAMPLE_SIZE, hexdigest=False):
+    #get file size from file object
```

Comment:
Have you tried this pattern: `size = os.fstat(f.fileno()).st_size`

I've not used it but I've seen it cited widely.

Reply:
i've used that pattern for checking the type of file on other projects, but not for the size. it should work fine though. ill make the change.